### PR TITLE
feat: adds support for access rules on applications

### PR DESCRIPTION
**Pull request by dhudec**
## Description

- Adds support for access rules on applications
- Upgrades basistheory-go library to v3.7.0

## Testing required outside of automated testing?

- [x] Not Applicable

### Screenshots (if appropriate):

- [x] Not Applicable

## Rollback / Rollforward Procedure

- [x] Roll Forward
- [ ] Roll Back

## Reviewer Checklist

- [ ] Description of Change
- [ ] Description of outside testing if applicable.
- [ ] Description of Roll Forward / Backward Procedure
- [ ] Documentation updated for Change

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# Contributing
```
Comment (djejaquino) [inline CONTRIBUTING.md:1]: Nice addition

File: docs/resources/application.md [hunk 2]
```diff
@@ -37,9 +50,13 @@ output "application_key" {
 ### Required
 
 - `name` (String) Name of the Application
-- `permissions` (Set of String) Permissions for the Application
 - `type` (String) Type for the Application
 
+### Optional
```
Comment (djejaquino) [inline docs/resources/application.md:55]: Is one of the below required?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+The examples included under `/examples/resources` should be manually updated
+with any new resources that are introduced.
+
+After updating the examples, reformat and regenerate the markdown docs under 
+`/docs/resources` by running the commands:
+
+```shell
+terraform fmt -recursive ./examples/
+go run github.com/hashicorp/terraform-plugin-docs/cmd/tfplugindocs
```
Comment (brigonzalez) [inline CONTRIBUTING.md:31]:
```suggestion
go generate
```
`go generate` will run both of these commands. this is configured in `main.go` btw

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+make verify
+```
+
+## Updating examples
+
+The examples included under `/examples/resources` should be manually updated
+with any new resources that are introduced.
+
+After updating the examples, reformat and regenerate the markdown docs under 
+`/docs/resources` by running the commands:
```
Comment (brigonzalez) [inline CONTRIBUTING.md:27]:
```suggestion
`/docs/resources` by running the command:
```

File: docs/resources/application.md (hunk 2 above)
Comment (dhudec) [reply docs/resources/application.md:55]: Either rules or permissions is required, but I don't think we can express that through the terraform provider. A 400 error will be returned if a user tries to create an app with neither, or with both.

**Review by brigonzalez — APPROVED** [review]

**Pull request merged**